apps/finance/app/src/components/Transfers.js

```diff
@@ -109,24 +62,13 @@ const Transfers = React.memo(({ tokens, transactions }) => {
     },
     {}
   )
-  const { resolve: resolveAddress } = useContext(IdentityContext)
-  const handleDownload = useCallback(async () => {
-    if (!currentApp || !currentApp.appAddress) {
-      return
-    }
 
-    const data = await getDownloadData(
-      filteredTransfers,
-      tokenDetails,
-      resolveAddress
-    )
-    const filename = getDownloadFilename(
-      currentApp.appAddress,
-      selectedDateRange
-    )
-    saveAs(new Blob([data], { type: 'text/csv;charset=utf-8' }), filename)
-    toast('Transfers data exported')
-  }, [currentApp, filteredTransfers, tokenDetails, resolveAddress])
+  const { handleDownload } = useDownloadData({
+    filteredTransfers,
+    selectedDateRange,
+    toast,
```

Comment:
Rather than providing this, we should directly use `useToast()` in the `useDownload()` hook.